Assets/Scripts/CameraFollow.cs

```diff
@@ -6,15 +6,17 @@
 
 public class CameraFollow : MonoBehaviour
 {
-	public float maxSpeed = 10f;
 	public float xMargin = 1f; // Distance in the x axis the player can move before the camera follows.
     public float yMargin = 1f; // Distance in the y axis the player can move before the camera follows.
     public float xSmooth = 8f; // How smoothly the camera catches up with it's target movement in the x axis.
     public float ySmooth = 8f; // How smoothly the camera catches up with it's target movement in the y axis.
     public Vector2 maxXAndY; // The maximum x and y coordinates the camera can have.
     public Vector2 minXAndY; // The minimum x and y coordinates the camera can have.
 
+    private float maxSpeed;
```

Comment:
sorry jay, I think I gave you a really bad advice, camerafollow should only care about the current player's max speed, it shouldn't do any of the calculation nor find out what the max speed is from the doll manager